test/iaas.AzureClient.spec.js

```diff
@@ -0,0 +1,366 @@
+                </Metadata>  
+              </Blob>
+            </Blobs>  
+            <NextMarker />   
+          </EnumerationResults>`
+        };
+        mocks.azureClient.list(settings.container, undefined, expectedResponse);
+        return client.list()
+          .then(files => {
+            expect(files).to.have.length(blobList.length);
+            expect(files[0].name).to.eql(blobList[0]);
+            expect(files[1].name).to.eql(blobList[1]);
+            expect(files[0].isTruncated).to.eql(false);
+            expect(files[1].isTruncated).to.eql(false);
+            mocks.verify();
+          });
+      });
+
+      it('blob upload should be successful', function () {
+        mocks.azureClient.upload(`/${settings.container}/${blobName}`, {
+          status: 201,
+          headers: {
+            'transfer-encoding': 'chunked',
+            'content-md5': '+v4bYMJBB8zY9FYiE+RISQ==',
+            'last-modified': 'Thu, 05 Jan 2017 03:13:04 GMT',
+            etag: '"0x8D43518C381D87C"',
+            server: 'Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0',
+            'x-ms-request-id': '5dc67ddb-0001-003d-7701-672321000000',
+            'x-ms-version': '2016-05-31',
+            'x-ms-request-server-encrypted': 'false',
+            date: new Date().toISOString(),
+            connection: 'close'
+          },
+          body: scheduled_data
+        }, 2);
+        return client.uploadJson(blobName, scheduled_data)
+          .then(result => {
+            expect(result).to.eql(scheduled_data);
+            mocks.verify();
+          });
+      });
+
+      it('blob upload should fail: PreconditionFailed', function () {
+        mocks.azureClient.upload(`/${settings.container}/${blobName}`, {
+          status: 412,
+          body: '<?xml version=\"1.0\" encoding=\"utf-8\"?><Error><Code>ConditionNotMet</Code><Message>The condition specified using HTTP conditional header(s) is not met.\nRequestId:879cee1d-0001-003e-4001-672026000000\nTime:2017-01-05T03:13:51.1418462Z</Message></Error>',
+          headers: {
+            'content-length': '252',
+            'content-type': 'application/xml',
+            server: 'Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0',
+            'x-ms-request-id': '879cee1d-0001-003e-4001-672026000000',
+            'x-ms-version': '2016-05-31',
+            date: 'Thu, 05 Jan 2017 03:13:50 GMT',
+            connection: 'close'
+          }
+        });
+        return client.uploadJson(blobName, scheduled_data)
+          .then(() => {
+            throw new Error('Upload data should have thrown error');
+          })
+          .catch(err => {
+            expect(err.statusCode || err.code).to.eql(412);
+            mocks.verify();
+          });
+
+      });
+
+      it('blob download should be successful', function () {
+        mocks.azureClient.download(`/${settings.container}/${blobName}`, scheduled_data, {
+          'content-type': 'application/octet-stream',
```

Comment:
Are all these response headers required? Most of the param values wont have right values. So if not required, please remove them.

Reply:
removed many headers.